include/mgp.hpp

```diff
@@ -3185,6 +3187,44 @@ inline bool Value::IsDuration() const { return mgp::value_is_duration(ptr_); }
 inline bool Value::operator==(const Value &other) const { return util::ValuesEqual(ptr_, other.ptr_); }
 
 inline bool Value::operator!=(const Value &other) const { return !(*this == other); }
+
+inline bool Value::operator<(const Value &other) const {
+  const mgp::Type &type = Type();
+  if (type != other.Type() && !(IsNumeric() && other.IsNumeric())) {
+    throw ValueException("Values have to be of the same type");
+  }
+
+  switch (type) {
+    case Type::Null:
+      throw ValueException("Cannot compare Null types");
+    case Type::Bool:
+      return ValueBool() < other.ValueBool();
+    case Type::Int:
+      return ValueNumeric() < other.ValueNumeric();
+    case Type::Double:
+      return ValueNumeric() < other.ValueNumeric();
+    case Type::String:
+      return ValueString() < other.ValueString();
+    case Type::Node:
+      return ValueNode() < other.ValueNode();
+    case Type::Relationship:
+      return ValueRelationship() < other.ValueRelationship();
+    case Type::Date:
+      return ValueDate() < other.ValueDate();
+    case Type::LocalTime:
+      return ValueLocalTime() < other.ValueLocalTime();
+    case Type::LocalDateTime:
+      return ValueLocalDateTime() < other.ValueLocalDateTime();
+    case Type::Duration:
+      return ValueDuration() < other.ValueDuration();
+    default:
+      if (IsPath() || IsList() || IsMap()) {
+        throw ValueException("Operator < is not defined for this data type");
+      } else {
+        throw ValueException("Undefined behaviour");
+      }
```

Comment:
```suggestion
    case Type::Path:
    case Type::List:
    case Type::Map:
        throw ValueException("Operator < is not defined for this Path, List or Map data type");
     default:
        throw ValueException("Undefined behaviour");
```

You can use this concept of fail through in switch case, and then default for this error message. This is only comment left from me